_posts/2016-02-10-installing-server-environment-with-powershell.md

```diff
@@ -0,0 +1,199 @@
+
+#### Windows features (e.g. IIS)
+The first question you should be looking is "what are the windows features"? To be able to install the features you must first learn how to find them. Here is a oneliner that will give the available features to you. 
+
+```powershell
+Get-WindowsOptionalFeature -Online 
+```
+
+Great! Now we know how to find the windows features. Now we just need to change the oneliner a bit to be able to Install them. This is a oneliner for .NET:  
+
+```powershell
+Enable-WindowsOptionalFeature -Online -All -FeatureName NetFx4
+```
+
+All flag means that it will also install required dependencies if any. The .NET delivered among the operating system might not be latest but the feature needs to be enabled all the same. IIS will need it among some other features. Full list of my users can be found in our example config file at the bottom of the article.
+
+#### Executables 
+With windows feature we were able to install the .NET version that is delivered with the operating system. Although that is not most likely the newest one. Instead we need to fetch the newest one from [the internet](https://download.microsoft.com/download/E/4/1/E4173890-A24A-4936-9FC9-AF930FE3FA40/NDP461-KB3102436-x86-x64-AllOS-ENU.exe). With executables you never can be quite sure how they should be installed because there is no stricly unified commandline parameters. Here is an example how your executable installation might looklike. 
+
+```powershell
+$arguments = @(
+        "/qn" # display no interface 
+        "/norestart"
+        "/passive"
+		"/S")
+Start-Process $exe  -Wait -PassThru -Verb runAs -ArgumentList $arguments
+```
+
+I chose an approach where I just guess what kind of parameters executable will need for silent installation. I found out that /S is pretty common, but sometimes also /passive is used too. Most likely your executables needs to be run elevated, that can be achieved with "-Verb runAs" parameter for Start-Process function. So even if this worked for my executables, this might be terrible mess for yours (for example if there is checks for arguments being eligible). 
+
+#### MSI files 
+MSI files are really common way to install stuff in Microsoft environment. They are also unified so we can be more certain that silent installation can be achieved. The installation looks really similar to installation of executables. 
+
+```powershell
+$arguments = @(
+        "/i" #install product
+        "`"$msiFile`""
+        "/qn" # display no interface 
+        "/norestart"
+        "/passive")
+Start-Process msiexec.exe -Wait -PassThru -Verb runAs -ArgumentList $arguments
+```
+
+File that is wanted to be installed is given as an parameter to msiexec.exe among some other flags. Installer is again run as elevated process to make sure that it can make all the configuration changes it wants. 
```

Comment:
File that is wanted to be installed -> The file that needs to be installed. Tai vielä lyhyemmin: The file to be installed